### PR TITLE
add high version kernel support (ipv4 and ipv6)

### PR DESCRIPTION
**Pull request by DavidLeeUX**
Test Environment:
kernel: 4.13.3-1.el7.elrepo.x86_64
kernel: 4.9.2.el7.centos.x86_64
kernel: 3.10.0-229.el7.x86_64

### CODE REVIEW

File: kmod/toa/toa.c [hunk 1]
```diff
@@ -341,16 +365,31 @@ tcp_v6_sk_destruct_toa(struct sock *sk) {
 	inet_sock_destruct(sk);
 }
 
+
```
Comment (mscbg) [inline kmod/toa/toa.c]: delete blank line

File: kmod/toa/toa.c [hunk 2]
```diff
@@ -380,13 +419,23 @@ tcp_v6_syn_recv_sock_toa(struct sock *sk, struct sk_buff *skb,
...
+        unsigned int level;
+	pte_t *pte;
 	/* hook inet_getname for ipv4 */
 	struct proto_ops *inet_stream_ops_p =
 			(struct proto_ops *)&inet_stream_ops;
 	/* hook tcp_v4_syn_recv_sock for ipv4 */
 	struct inet_connection_sock_af_ops *ipv4_specific_p =
 			(struct inet_connection_sock_af_ops *)&ipv4_specific;
 
+
```
Comment (mscbg) [inline kmod/toa/toa.c]: delete redundant blank line

File: kmod/toa/toa.c [hunk 3]
```diff
@@ -407,20 +456,34 @@ hook_toa_functions(void)
 		ipv6_specific_p->syn_recv_sock);
 #endif
 
+
+	pte = lookup_address((unsigned long )inet_stream_ops_p, &level);
```
Comment (mscbg) [inline kmod/toa/toa.c]: delete redundant blank line

**Comment by ywc689** [comment]
Please fix the conflicts.

**Comment by ywc689** [comment]
pr #484 , pr #571 had resolved the problem. Thanks for your contribution anyway.

**Pull request closed without merging**